src/c++/perf_analyzer/test_perf_utils.cc

```diff
@@ -114,4 +119,254 @@ TEST_CASE("testing the ParseTensorFormat function")
+  std::string temp_path{"/tmp/abcdef1234"};
+
+  CHECK(!IsDirectory(temp_path));
+
+  status = mkdir(temp_path.c_str(), S_IRWXU | S_IROTH | S_IXOTH);
+  REQUIRE(status == 0);
+  CHECK(IsDirectory(temp_path));
+
+  status = rmdir(temp_path.c_str());
+  REQUIRE(status == 0);
+  CHECK(!IsDirectory(temp_path));
+}
+
+TEST_CASE("perf_utils: IsFile")
+{
+  // Create a temporary file /tmp/test.txt
+  int status;
+  std::string temp_path{"/tmp/test.txt"};
+
+  CHECK(!IsFile(temp_path));
+
+  std::ofstream file(temp_path);
+  CHECK(IsFile(temp_path));
+
+  std::remove(temp_path.c_str());
+  CHECK(!IsFile(temp_path));
+}
+
+TEST_CASE("perf_utils: ByteSize")
+{
+  std::vector<int64_t> shape{3, 4, 5};
+
+  SUBCASE("Single byte elements")
+  {
+    CHECK(ByteSize(shape, "BOOL") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "INT8") == 1 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT8") == 1 * 3 * 4 * 5);
+  }
+
+  SUBCASE("2 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP16") == 2 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "BF16") == 2 * 3 * 4 * 5);
+  }
+
+  SUBCASE("4 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT32") == 4 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP32") == 4 * 3 * 4 * 5);
+  }
+
+  SUBCASE("8 byte elements")
+  {
+    CHECK(ByteSize(shape, "INT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "UINT64") == 8 * 3 * 4 * 5);
+    CHECK(ByteSize(shape, "FP64") == 8 * 3 * 4 * 5);
+  }
+
+  SUBCASE("Dynamic shape tensor")
+  {
+    shape.insert(shape.begin(), -1);
+
+    CHECK(ByteSize(shape, "BOOL") == -1);
+    CHECK(ByteSize(shape, "INT8") == -1);
+    CHECK(ByteSize(shape, "UINT8") == -1);
+
+    CHECK(ByteSize(shape, "INT16") == -1);
+    CHECK(ByteSize(shape, "UINT16") == -1);
+    CHECK(ByteSize(shape, "FP16") == -1);
+    CHECK(ByteSize(shape, "BF16") == -1);
+
+    CHECK(ByteSize(shape, "INT32") == -1);
+    CHECK(ByteSize(shape, "UINT32") == -1);
+    CHECK(ByteSize(shape, "FP32") == -1);
+
+    CHECK(ByteSize(shape, "INT64") == -1);
+    CHECK(ByteSize(shape, "UINT64") == -1);
+    CHECK(ByteSize(shape, "FP64") == -1);
+  }
+
+  SUBCASE("Unknown data types")
+  {
+    CHECK(ByteSize(shape, "bool") == -1);
+    CHECK(ByteSize(shape, "int8") == -1);
+    CHECK(ByteSize(shape, "uint8") == -1);
+
+    CHECK(ByteSize(shape, "int16") == -1);
+    CHECK(ByteSize(shape, "uint16") == -1);
+    CHECK(ByteSize(shape, "fp16") == -1);
+    CHECK(ByteSize(shape, "bf16") == -1);
+
+    CHECK(ByteSize(shape, "int32") == -1);
+    CHECK(ByteSize(shape, "uint32") == -1);
+    CHECK(ByteSize(shape, "fp32") == -1);
+
+    CHECK(ByteSize(shape, "int64") == -1);
+    CHECK(ByteSize(shape, "uint64") == -1);
+    CHECK(ByteSize(shape, "fp64") == -1);
+
+    CHECK(ByteSize(shape, "abc") == -1);
+    CHECK(ByteSize(shape, "1234") == -1);
+    CHECK(ByteSize(shape, "") == -1);
+  }
+}
```

Comment:
Should we differentiate return values between a dynamic shape and an unknown datatype? Both return same `-1` currently. 

Currently the "unknown data type" subtest could just be computing the shape in the same way as in the dynamic test since `shape` still contains a `-1` in this subtest.